ipaserver/install/plugins/update_ra_cert_store.py

```diff
@@ -0,0 +1,115 @@
+# Authors:
+#   Alexander Bokovoy <abokovoy@redhat.com>
+#
+# Copyright (C) 2012  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import binascii
+import os
+import pwd
+
+from ipalib import Registry
+from ipalib import Updater
+from ipalib.install import certmonger
+from ipaplatform.constants import constants
+from ipaplatform.paths import paths
+from ipaplatform.tasks import tasks
+from ipapython import certdb
+
+register = Registry()
+
+
+@register()
+class update_ra_cert_store(Updater):
+    """
+    Moves the cert store from /etc/httpd/alias to /var/lib/ipa/radb
+    """
+
+    def execute(self, **options):
+        olddb = certdb.NSSDatabase(nssdir=paths.HTTPD_ALIAS_DIR)
+        if not olddb.has_nickname('ipaCert'):
+            # Nothign to do
+            return False, []
+
+        newdb = certdb.NSSDatabase(nssdir=paths.IPA_RADB_DIR)
+        if os.path.exists(paths.IPA_RADB_DIR):
+            if newdb.has_nickname:
+                self.log.warning(
+                    "An 'ipaCert' nickname exists in both the old {} and the "
+                    "new {} NSS Databases!".format(paths.HTTPD_ALIAS_DIR,
+                                                   paths.IPA_RADB_DIR))
+                return False, []
+        else:
+            # Get the owning user ids
+            pent = pwd.getpwnam(constants.IPAAPI_USER)
+
+            # create the initial directory
+            os.makedirs(paths.IPA_RADB_DIR, 0o751)
+            os.chown(paths.IPA_RADB_DIR, pent.pw_uid, pent.pw_gid)
+            tasks.restore_context(paths.IPA_RADB_DIR)
+
+            # Create the password file for this db
+            hex_str = binascii.hexlify(os.urandom(10))
+            f = os.open(paths.IPA_RADB_PWDFILE_TXT, os.O_CREAT | os.O_RDWR)
+            os.write(f, hex_str)
+            os.close(f)
+
+            # And create the DB
+            newdb.create_db(paths.IPA_RADB_PWDFILE_TXT)
```

Comment:
I think there was a reason why I could not reuse the same code, but I'll investigate merging these if I can move that stuff to the CA Instance code.